### PR TITLE
Refactor/interface world new

### PR DESCRIPTION
**Pull request by emielsteerneman**
The old world implementation should be removed from the interface everywhere. There is still some stuff in Output, but I don't know what to do with that, because it's filled with statics

### CODE REVIEW

**Comment by emielsteerneman** [comment]
There are still some parts in widget.cpp that are not yet implemented, because these things are not yet in RobotView

File: include/roboteam_ai/interface/widgets/widget.h [hunk 1]
```diff
@@ -41,25 +38,26 @@ class Visualizer : public QWidget {
...
     void setToggleFieldDirection(bool inversed);
 
    protected:
     void paintEvent(QPaintEvent *event) override;
     void mousePressEvent(QMouseEvent *event) override;
 
    private:
-    float factor;
+    const rtt::world_new::World &worldManager;
+    float factor{};
```
Comment (rolfvdhulst) [inline include/roboteam_ai/interface/widgets/widget.h:50]: wtf is this bracket initialization? Have never seen this before.

File: src/roboteam_ai.cpp [hunk 2]
```diff
@@ -77,7 +78,10 @@ int main(int argc, char *argv[]) {
     // initialize the interface
     QApplication a(argc, argv);
     setDarkTheme();
-    window = std::make_shared<ui::MainWindow>();
+
+    // Todo make this a not-global-static thingy
+    rtt::world_new::World* worldManager = rtt::world_new::World::instance();
+    window = std::make_shared<ui::MainWindow>(*worldManager);
     window->setWindowState(Qt::WindowMaximized);
```
Comment (rolfvdhulst) [inline src/roboteam_ai.cpp:85]: Can't you const ref from application manager? Application manager should also own an instance. Having 2 instances is quite ugly...

File: src/interface/widgets/widget.cpp [hunk 3]
```diff
@@ -258,15 +260,15 @@ void Visualizer::drawBall(QPainter &painter) {
 }
 
 // draw the robots
-void Visualizer::drawRobots(QPainter &painter) {
+void Visualizer::drawRobots(QPainter &painter, rtt::world_new::view::WorldDataView world) {
     // draw us
-    for (auto &robot : rtt::ai::world::world->getUs()) {
-        drawRobot(painter, *robot, true);
+    for (rtt::world_new::view::RobotView robot : world->getUs()) {
```
Comment (rolfvdhulst) [inline src/interface/widgets/widget.cpp:265]: const &?

Comment (rolfvdhulst) [reply src/interface/widgets/widget.cpp:265]: oh wait nvm these things are pointers right? Not sure.

File: src/interface/widgets/widget.cpp [hunk 4]
```diff
@@ -337,68 +337,42 @@ void Visualizer::drawRobot(QPainter &painter, Robot robot, bool ourTeam) {
         painter.drawText(robotpos.x, ypos += 20, QString::fromStdString(getRoleNameForRobot(robot)));
     }
 
+    // Todo : Get working stuff in RobotView
```
Comment (rolfvdhulst) [inline src/interface/widgets/widget.cpp:340]: ?

File: src/interface/widgets/widget.cpp [hunk 5]
```diff
@@ -337,68 +337,42 @@ void Visualizer::drawRobot(QPainter &painter, Robot robot, bool ourTeam) {
...
-                painter.setBrush(Qt::green);
-            } else {
-                // yellow to indicate feedback is not okay
-                painter.setPen(Qt::red);
-                painter.setBrush(Qt::red);
-            }
-            painter.drawEllipse({(int)robotpos.x + 10, (int)robotpos.y - 10}, 2, 2);
-        }
-    }
+    // Todo : Get working feedback in RobotView
```
Comment (rolfvdhulst) [inline src/interface/widgets/widget.cpp:356]: Way to do this is by making feedback processing a roboteam_world thing. Make this an issue (not critical for now).

**Review by rolfvdhulst — COMMENTED** [review]
I looked at all the non-linter changes and they look pretty decent. 

You might want to re-enable all of the prints you commented as it might mess up people who expect them to be printed.

File: src/roboteam_ai.cpp (hunk 2 above)
Comment (rolfvdhulst) [reply src/roboteam_ai.cpp:85]: Maybe I am understanding it wrong here though.

File: include/roboteam_ai/interface/widgets/widget.h (hunk 1 above)
Comment (emielsteerneman) [reply include/roboteam_ai/interface/widgets/widget.h:50]: Ah, clang did that for me. It was complaining that it was uninitialized

File: src/interface/widgets/widget.cpp (hunk 3 above)
Comment (emielsteerneman) [reply src/interface/widgets/widget.cpp:265]: They are RobotViews, which only hold a single pointer to actual robot data. They wouldn't have to be const but I'll add it anyway for consistency

File: src/interface/widgets/widget.cpp (hunk 4 above)
Comment (emielsteerneman) [reply src/interface/widgets/widget.cpp:340]: Woops, forgot to remove that

File: src/roboteam_ai.cpp (hunk 2 above)
Comment (emielsteerneman) [reply src/roboteam_ai.cpp:85]: I'm getting the global instance here. There is always exactly one instance. Always.

File: src/interface/widgets/widget.cpp (hunk 5 above)
Comment (emielsteerneman) [reply src/interface/widgets/widget.cpp:356]: Issue made

**Review by rolfvdhulst — APPROVED** [review]

**Pull request merged**